crates/burn-fusion/src/tensor.rs

```diff
@@ -133,6 +128,21 @@ impl<R: FusionRuntime> FusionTensor<R> {
             .await
     }
 
+    pub(crate) async fn q_into_data<B>(self) -> TensorData
+    where
+        B: FusionBackend<FusionRuntime = R>,
+    {
+        if let DType::QFloat(_scheme) = self.dtype {
+            let id = self.stream;
+            self.client
+                .clone()
+                .read_tensor_quantized::<B>(self.into_description(), id)
+                .await
+        } else {
+            panic!("Expected quantized float dtype, got {:?}", self.dtype)
+        }
+    }
```

Comment:
We should not call await here. it creates deadlocks.

```rust
pub(crate) fn q_into_data<B>(self) -> impl Future<Output=TensorData>
    where
        B: FusionBackend<FusionRuntime = R>,
    {
        if let DType::QFloat(_scheme) = self.dtype {
            let id = self.stream;
            let fut = self.client
                .clone()
                .read_tensor_quantized::<B>(self.into_description(), id);
                async move {
                    fut.await
                }
        } else {
            panic!("Expected quantized float dtype, got {:?}", self.dtype)
        }
    }
 ```

Reply:
Changed all other `into_data` methods for `FusionTensor` 👍